crates/core_simd/src/reduction.rs

```diff
@@ -0,0 +1,162 @@
+macro_rules! impl_integer_reductions {
+    { $name:ident, $scalar:ty } => {
+        impl<const LANES: usize> crate::$name<LANES>
+        where
+            Self: crate::LanesAtMost32
+        {
+            /// Horizontal wrapping add.  Returns the sum of the lanes of the vector, with wrapping addition.
+            #[inline]
+            pub fn horizontal_wrapping_sum(self) -> $scalar {
+                unsafe { crate::intrinsics::simd_reduce_add_ordered(self, 0) }
+            }
+
+            /// Horizontal wrapping multiply.  Returns the product of the lanes of the vector, with wrapping multiplication.
+            #[inline]
+            pub fn horizontal_wrapping_product(self) -> $scalar {
+                unsafe { crate::intrinsics::simd_reduce_mul_ordered(self, 1) }
+            }
```

Comment:
Wrapping is implied throughout the crate by default, is it not?

Reply:
For the ops, yes, but I figure it should be explicit for any named functions still

Reply:
I lean toward "don't say wrapping in the name, it's the default and people should just learn that".

While saying it every time helps the small portion of people who are new to SIMD usage, it becomes a bother to people who are past the initial introduction.

Reply:
Yeah I agree this is probably right, there are no non-wrapping functions to distinguish from so it's unnecessary.